lib/metadata/schema-parser.js

```diff
@@ -1027,14 +1027,17 @@ function getTokenToReplicaSimpleMapper(replicationFactor) {
     for (let i = 0; i < ringLength; i++) {
       const key = ringTokensAsStrings[i];
       const tokenReplicas = [primaryReplicas[key]];
-      for (let j = 1; j < rf; j++) {
+      for (let j = 1; j < ringLength && tokenReplicas.length < rf; j++) {
         let nextReplicaIndex = i + j;
         if (nextReplicaIndex >= ringLength) {
           //circle back
           nextReplicaIndex = nextReplicaIndex % ringLength;
         }
         const nextReplica = primaryReplicas[ringTokensAsStrings[nextReplicaIndex]];
-        tokenReplicas.push(nextReplica);
+        // In the case of vnodes, consecutive sections of the ring can be assigned to the same host.
+        if (tokenReplicas.indexOf(nextReplica) === -1) {
+          tokenReplicas.push(nextReplica);
```

Comment:
Nice catch and fix!
We could maybe benefit from using ES6 `Set` here but it's out of scope of this feature. I've created a separate ticket for it to be considered in the future : NODEJS-420.